pkg/validations/createvalidations/gitops.go

```diff
@@ -28,6 +36,24 @@ func ValidateGitOps(ctx context.Context, k validations.KubectlClient, cluster *t
 	if err != nil {
 		return fmt.Errorf("workload cluster gitOpsConfig is invalid: %v", err)
 	}
+
+	return err
+}
+
+func validateAuthenticationGitProvider(clusterSpec *cluster.Spec, cliConfig *config.CliConfig) error {
+	fluxConfig := clusterSpec.FluxConfig
+	if fluxConfig.Spec.Git == nil {
+		return nil
+	}
+	if cliConfig.GitPassword == "" && cliConfig.GitPrivateKeyFile == "" {
+		return fmt.Errorf("for git provider in the Flux config either set a password using %s env var or path to private key"+
+			" file using %s", config.EksaGitPasswordTokenEnv, config.EksaGitPrivateKeyTokenEnv)
+	}
+	if cliConfig.GitPrivateKeyFile != "" {
```

Comment:
If both are set, what happens? This is mostly for my understanding, I'm not asking to change anything. Just wondering if we need a validation or if just one of them takes precedence (which we could print a warning or something like that)

Reply:
If both are set, we pass them both as params without any one taking specific precendence
https://fluxcd.io/docs/cmd/flux_bootstrap_git/#examples